step.go

```diff
@@ -0,0 +1,484 @@
+		fmt.Println()
+		log.Printf("Installing xcpretty")
+
+		cmdModelSlice, err := formatter.Install()
+		if err != nil {
+			return fmt.Errorf("failed to create xcpretty install commands: %w", err)
+		}
+
+		for _, cmd := range cmdModelSlice {
+			if err := cmd.Run(); err != nil {
+				return fmt.Errorf("failed to run xcpretty install command (%s): %w", cmd.PrintableCommandArgs(), err)
+			}
+		}
+	}
+
+	xcprettyVersion, err := formatter.Version()
+	if err != nil {
+		return fmt.Errorf("failed to get xcpretty version: %w", err)
+	}
+
+	log.Printf("- xcpretty version: %s", xcprettyVersion.String())
+	return nil
+}
+
+type RunOut struct {
+	XcodebuildLog string
+	BuiltTestDir  string
+	XctestrunPth  string
+	SYMRoot       string
+}
+
+func (b TestBuilder) Run(cfg Config) (RunOut, error) {
+	// Automatic code signing
+	authOptions, err := b.automaticCodeSigning(cfg.CodesignManager)
+	if err != nil {
+		return RunOut{}, err
+	}
+
+	// Build for testing
+	log.Infof("Build:")
+
+	xcconfigWriter := xcconfig.NewWriter(v2pathutil.NewPathProvider(), v2fileutil.NewFileManager())
+	xcconfigPath, err := xcconfigWriter.Write(cfg.XCConfigContent)
+	if err != nil {
+		return RunOut{}, err
+	}
+
+	xcodeBuildCmd := xcodebuild.NewCommandBuilder(cfg.ProjectPath, xcworkspace.IsWorkspace(cfg.ProjectPath), "")
+	xcodeBuildCmd.SetScheme(cfg.Scheme)
+	xcodeBuildCmd.SetConfiguration(cfg.Configuration)
+	xcodeBuildCmd.SetCustomBuildAction("build-for-testing")
+	xcodeBuildCmd.SetDestination(cfg.Destination)
+	xcodeBuildCmd.SetCustomOptions(cfg.XcodebuildOptions)
+	xcodeBuildCmd.SetXCConfigPath(xcconfigPath)
+	if authOptions != nil {
+		xcodeBuildCmd.SetAuthentication(*authOptions)
+	}
+
+	result := RunOut{}
+	rawXcodebuildOut, buildInterval, err := runCommandWithRetry(xcodeBuildCmd, cfg.XCPretty, cfg.SwiftPackagesPath)
+	if err != nil || !cfg.XCPretty {
+		printLastLinesOfXcodebuildTestLog(rawXcodebuildOut, err == nil)
+	}
+
+	result.XcodebuildLog = rawXcodebuildOut
+
+	if err != nil {
+		return result, err
+	}
+
+	// Cache swift packages
+	if cfg.CacheLevel == "swift_packages" {
+		if err := cache.CollectSwiftPackages(cfg.ProjectPath); err != nil {
+			log.Warnf("Failed to mark swift packages for caching: %s", err)
+		}
+	}
+
+	// Find outputs
+	testBundle, err := b.findTestBundle(findTestBundleOpts{
+		ProjectPath:       cfg.ProjectPath,
+		Scheme:            cfg.Scheme,
+		Configuration:     cfg.Configuration,
+		XcodebuildOptions: cfg.XcodebuildOptions,
+		BuildInterval:     buildInterval,
+	})
+	if err != nil {
+		return result, err
+	}
+
+	result.BuiltTestDir = testBundle.BuiltTestDir
+	result.XctestrunPth = testBundle.XctestrunPth
+	result.SYMRoot = testBundle.SYMRoot
+
+	return result, nil
+}
+
+type ExportOpts struct {
+	RunOut
+	OutputDir string
+}
+
+func (b TestBuilder) ExportOutput(opts ExportOpts) error {
+	log.Infof("Export:")
+
+	if opts.XcodebuildLog != "" {
+		if err := exportXcodebuildTestLog(opts.OutputDir, opts.XcodebuildLog); err != nil {
```

Comment:
Is it worth printing the location of the raw build log? In some cases printLastLinesOfXcodebuildTestLog mentions the env var, but not in all cases.

Reply:
For the sake of simplicity, I would print all the exported outputs in this phase.